spec/services/articles/feeds_spec.rb

```diff
@@ -0,0 +1,44 @@
+require "rails_helper"
+
+RSpec.describe Articles::Feeds do
+  describe ".oldest_published_at_to_consider_for" do
+    subject(:function_call) { described_class.oldest_published_at_to_consider_for(user: user) }
+
+    context "when the given user is nil" do
+      let(:user) { nil }
+
+      it { is_expected.to be_a ActiveSupport::TimeWithZone }
+    end
+
+    context "when the given user has no page views" do
+      let(:user) { instance_double(User) }
+
+      before do
+        allow(user).to receive(:page_views).and_return(nil)
```

Comment:
A user having no page views returns an empty array rather than `nil`. Doesn't look like [the application code](https://github.com/forem/forem/pull/15528/files#diff-ee753227442f6c30328ea8b6e02966f4359d3c5a92d35f2f7261967f3f3676aeR35) needs to be updated for this change.

```suggestion
        allow(user).to receive(:page_views).and_return([])
```